### PR TITLE
James 2754 part 1 -  Add smtp Port parameter to deploiement validation test

### PR DESCRIPTION
**Pull request by remk**
First step toward https://github.com/linagora/james-project/pull/2334.

Need to be merged at the same time as : 
https://github.com/linagora/james-jenkins/pull/152

### CODE REVIEW

File: mpt/impl/imap-mailbox/external-james/pom.xml [hunk 1]
```diff
@@ -39,6 +39,10 @@
             <groupId>${james.groupId}</groupId>
             <artifactId>apache-james-mpt-core</artifactId>
         </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-testing</artifactId>
```
Comment (trantienduchn) [inline mpt/impl/imap-mailbox/external-james/pom.xml]: test scope?

File: mpt/impl/imap-mailbox/external-james/src/test/java/org/apache/james/mpt/imapmailbox/external/james/host/ExternalJamesConfigurationEnvironnementVariables.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    private static final String ENV_JAMES_SMTP_PORT = "JAMES_SMTP_PORT";
+
+    private final String address;
+    private final Port imapPort;
+    private final Port smtpPort;
+
+    public ExternalJamesConfigurationEnvironnementVariables() {
+        Preconditions.checkState(System.getenv(ENV_JAMES_ADDRESS) != null, "You must have exported an environment variable called JAMES_ADDRESS in order to run these tests. For instance export JAMES_ADDRESS=127.0.0.1");
+        Preconditions.checkState(System.getenv(ENV_JAMES_IMAP_PORT) != null, "You must have exported an environment variable called JAMES_IMAP_PORT in order to run these tests. For instance export JAMES_IMAP_PORT=143");
+        Preconditions.checkState(System.getenv(ENV_JAMES_SMTP_PORT) != null, "You must have exported an environment variable called JAMES_SMTP_PORT in order to run these tests. For instance export JAMES_IMAP_PORT=143");
```
Comment (trantienduchn) [inline mpt/impl/imap-mailbox/external-james/src/test/java/org/apache/james/mpt/imapmailbox/external/james/host/ExternalJamesConfigurationEnvironnementVariables.java]: s/`For instance export JAMES_IMAP_PORT=143`/`For instance export JAMES_SMTP_PORT=25`

File: mpt/impl/imap-mailbox/external-james/src/test/java/org/apache/james/mpt/imapmailbox/external/james/host/ExternalJamesSmtpHostSystem.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+public class ExternalJamesSmtpHostSystem implements SmtpHostSystem{
+
+    private final ExternalJamesConfiguration configuration;
+
+    @Inject
+    private ExternalJamesSmtpHostSystem(ExternalJamesConfiguration externalConfiguration) {
+        this.configuration = externalConfiguration;
+    }
+
+    public SMTPMessageSender connect(SMTPMessageSender smtpMessageSender) throws IOException {
```
Comment (trantienduchn) [inline mpt/impl/imap-mailbox/external-james/src/test/java/org/apache/james/mpt/imapmailbox/external/james/host/ExternalJamesSmtpHostSystem.java]: can you put `@Override` annotation here

**Comment by Arsnael** [comment]
```
[023aef4736fcd60f18d59dbb9d483e6af076c4c7] [ERROR] Errors: 
[023aef4736fcd60f18d59dbb9d483e6af076c4c7] [ERROR]   ForwardSmtpTest.forwardingAnEmailShouldWork:79 » ConditionTimeout Assertion co...
```
test this please

**Comment by remk** [comment]
test this please

**Pull request merged**